examples/megacli.cpp

```diff
@@ -6652,6 +6656,134 @@ void exec_enabletransferresumption(autocomplete::ACState& s)
     }
 }
 
+void exec_notifydrive(autocomplete::ACState& s)
+{
+#ifndef HAVE_DRIVE_NOTIFY
+    std::cout << "Failed! This functionality was disabled at compile time." << std::endl;
+#else
+
+    int timeout = 60; // seconds
+    bool showConnected = true;
+    bool showDisconnected = true;
+
+    // notifydrive [N|connect|disconnect]
+    if (s.words.size() == 2)
+    {
+        int t = 0;
+        try
+        {
+            t = stoi(s.words[1].s);
+        }
+        catch (...) {}
+
+        if (t)
+        {
+            timeout = t;
+        }
+        else
+        {
+            if (s.words[1].s == "connect")
+            {
+                showDisconnected = false;
+            }
+            else
+            {
+                showConnected = false;
+            }
+        }
+    }
+
+    // notifydrive [connect|disconnect] N
+    else if (s.words.size() == 3)
+    {
+        int t = 0;
+        try
+        {
+            t = stoi(s.words[2].s);
+        }
+        catch (...) {}
```

Comment:
How about the much shorter  `int t = atoi(...);` if we are going to ignore any exception anyway ;-)

Reply:
Or, much better - let the exception be thrown, and have a single catch(std::exception& e) in the main loop to report e.what()  for any command that throws such an exception?  thanks

Reply:
Actually the intended behavior was different than both suggested changes. It should indeed ignore errors and _continue_, using the default value, when an error has occurred.
`atoi()` does not throw exceptions, but it can cause undefined behavior.
By letting exception be thrown, it would still stop the execution of the command.
I can of course change the code the way you prefer; just pointing out the intended behavior.

Reply:
If the user supplied a parameter, we shouldn't silently ignore it - if we can't use it we should tell them there's been an error, either they mistyped or we misunderstood.